### PR TITLE
Bugfix: CRUD operations for folders

### PR DESCRIPTION
**Pull request by madsrasmussen**
This PR fixes the CRUD operations for folders for all trees. I had to do some changes for the rest of the tree crud operations too.

- Make sure the tree is updated when:
- creating  a folder
- renaming a folder
- deleting folder
- creating an item
- deleting an item
- updating an item

Known issue:
The rename script/stylesheet/patial-view entity action doesn't work. It will be fixed in another PR.

### CODE REVIEW

File: src/packages/core/tree/default/default-tree.context.ts [hunk 1]
```diff
@@ -52,7 +57,26 @@ export class UmbDefaultTreeContext<TreeItemType extends UmbTreeItemModelBase>
...
-		this.requestTreeRoot();
+
+		/* TODO: revisit. This is a temp solution to notify the parent it needs to reload its children
+		there might be a better way to do this through a tree item parent context.
+		It does not look like there is a way to have a "dynamic" parent context that will stop when a
+		specific parent is reached (a tree item unique that matches the parentUnique of this item) */
+		const hostElement = this.getHostElement();
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		hostElement.addEventListener('reload-tree-item-parent', (event: CustomEvent) => {
```
Comment (nielslyngsoe) [inline src/packages/core/tree/default/default-tree.context.ts:68]: Should we prefix the event name here.. with ´umb:....´

File: src/packages/data-type/workspace/data-type-workspace.context.ts [hunk 2]
```diff
@@ -283,6 +284,14 @@ export class UmbDataTypeWorkspaceContext
 			eventContext.dispatchEvent(event);
 		} else {
 			await this.repository.save(this.#currentData.value);
+
+			const actionEventContext = await this.getContext(UMB_ACTION_EVENT_CONTEXT);
```
Comment (nielslyngsoe) [inline src/packages/data-type/workspace/data-type-workspace.context.ts:288]: Why do we need to dispatch an event? who is the consumer of those events? .. :-) I'm thinking out loud here.

File: src/packages/core/tree/tree-item/tree-item-base/tree-item-context-base.ts [hunk 3]
```diff
@@ -78,6 +83,24 @@ export abstract class UmbTreeItemContextBase<TreeItemType extends UmbTreeItemMod
...
 		this.pagination.addEventListener(UmbChangeEvent.TYPE, this.#onPageChange);
+
+		/* TODO: revisit. This is a temp solution to notify the parent it needs to reload its children
+		there might be a better way to do this through a tree item parent context.
+		It does not look like there is a way to have a "dynamic" parent context that will stop when a
+		specific parent is reached (a tree item unique that matches the parentUnique of this item) */
+		const hostElement = this.getHostElement();
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		hostElement.addEventListener('reload-tree-item-parent', (event: CustomEvent) => {
```
Comment (nielslyngsoe) [inline src/packages/core/tree/tree-item/tree-item-base/tree-item-context-base.ts:94]: Make the name obviously temporary/hacky... so pepople dont start using it.

**Review by leekelleher — APPROVED** [review]
Tested out against the server API, all working great!

**Pull request merged**